xfer/buffer.go

```diff
@@ -2,40 +2,73 @@ package xfer
 
 import (
 	"bytes"
+	"io"
 	"sync"
 	"sync/atomic"
 )
 
 // A Buffer is a reference counted bytes.Buffer, which belongs
 // to a sync.Pool
-type Buffer struct {
+type Buffer interface {
+	io.Reader
+	io.Writer
```

Comment:
Also implements ByteReader and BytesWriter; could add it here to help with compression (see http://golang.org/src/compress/gzip/gunzip.go?s=2453:2497#L74) but I don't think its necessary.